tests/Kiota.Builder.Tests/Writers/TypeScript/CodeFunctionWriterTests.cs

```diff
@@ -1252,9 +1252,9 @@ public async Task Writes_UnionOfPrimitiveValues_SerializerFunctionAsync()
         writer.Write(serializerFunction);
         var serializerFunctionStr = tw.ToString();
         Assert.Contains("return", serializerFunctionStr);
-        Assert.Contains("switch", serializerFunctionStr);
-        Assert.Contains("case \"number\":", serializerFunctionStr);
-        Assert.Contains("case \"string\":", serializerFunctionStr);
+        Assert.Contains("switch (true)", serializerFunctionStr);
```

Comment:
why do we need a switch case here? Couldn't we rely on ifs instead? or use the typeof as a parameter for the switch?

Reply:
This is part of the code 

```typescript
export function serializeSuccess(writer: SerializationWriter, success: Partial<string[] | string> | undefined | null = {}) : void {
    if (success === undefined || success === null) return;
    switch (true) {
        case Array.isArray(success) && (success).every(item => typeof item === 'string') :
            writer.writeCollectionOfPrimitiveValues<string>(undefined, success as string[]);
            break;
        case typeof success === "string":
            writer.writeStringValue(undefined, success as string);
            break;
    }
}

```

I suppose it can be re-written as 

```typescript
export function serializeSuccess1(writer: SerializationWriter, success: Partial<string[] | string> | undefined | null = {}) : void {
    if (success === undefined || success === null) return;

    if(Array.isArray(success) && (success).every(item => typeof item === 'string')){
        writer.writeCollectionOfPrimitiveValues<string>(undefined, success as string[]);
    } else if (typeof success === "string") {
        writer.writeStringValue(undefined, success as string);
    }
}
```

it should be the same code, choose the earlier because it easier to read, subjective. We can't use the type in the switch statement as the type of all arrays will be an 'object'

Reply:
thanks for expanding on this. I feel like the later option will reduce the end JavaScript being emitted (switch) by a little. Which at scale will have an impact on the transpile time and assets size